nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/test/java/org/apache/nifi/processors/kudu/TestScanKudu.java

```diff
@@ -0,0 +1,426 @@
+        testRunner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
+    }
+
+    @Test
+    public void testKuduProjectedColumnsValidation() {
+        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
+        runner.setProperty(ScanKudu.PREDICATES, "column1=val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1,c2");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1 c2,c3");
+        runner.assertNotValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1:,c2,c3");
+        runner.assertNotValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1,c1,");
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testKuduPredicatesValidation() {
+        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
+        runner.setProperty(ScanKudu.PREDICATES, "column1=val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1>val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1<val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1<=val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1>=val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1=>=val1");
+        runner.assertNotValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1:val1");
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testNoIncomingFlowFile() {
+        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
+        runner.setProperty(ScanKudu.PREDICATES, "column1=val1");
+
+        runner.run(1, false);
+        runner.assertTransferCount(ScanKudu.REL_FAILURE, 0);
+        runner.assertTransferCount(ScanKudu.REL_SUCCESS, 0);
+        runner.assertTransferCount(ScanKudu.REL_ORIGINAL, 0);
+    }
+
+    @Test
+    public void testInvalidKuduTableName() throws KuduException {
+        final Map<String, String> rows = new HashMap<>();
+        rows.put("key", "val1");
+        rows.put("key1", "val1");
+
+        kuduScan.insertTestRecordsToKuduTable(DEFAULT_TABLE_NAME, rows);
+
+        runner.setProperty(ScanKudu.TABLE_NAME, "${table1}");
+        runner.setProperty(ScanKudu.PREDICATES, "key1=val1");
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "key1");
+        runner.setValidateExpressionUsage(false);
+        runner.enqueue("trigger flow file");
+        runner.run(1, false);
+
+        runner.assertTransferCount(ScanKudu.REL_FAILURE, 1);
+        runner.assertTransferCount(ScanKudu.REL_SUCCESS, 0);
+        runner.assertTransferCount(ScanKudu.REL_ORIGINAL, 0);
+
+    }
+
+    @Test
+    public void testInvalidPredicatesPattern() throws KuduException {
```

Comment:
Just to make clear up any potential confusion I'd rename this method and add a new one (as this doesn't test the _pattern_ of the predicate but the semantics of it - namely in this case it will fail because the referenced _column1_ doesn't exist).

So the 2 tests could look something like this:
```java
@Test
    public void testInvalidPredicateReferencingNonExistingColumn() throws KuduException {
        final Map<String, String> rows = new HashMap<>();
        rows.put("key", "val1");
        rows.put("key1", "val1");
        
        String patternReferencingNonExistingColumn = "column1=val1";

        kuduScan.insertTestRecordsToKuduTable(DEFAULT_TABLE_NAME, rows);

        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
        runner.setProperty(ScanKudu.PREDICATES, patternReferencingNonExistingColumn);
        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "column1");
        runner.setValidateExpressionUsage(false);
        runner.enqueue("trigger flow file");
        runner.run(1, false);

        runner.assertTransferCount(ScanKudu.REL_FAILURE, 1);
        runner.assertTransferCount(ScanKudu.REL_SUCCESS, 0);
        runner.assertTransferCount(ScanKudu.REL_ORIGINAL, 0);
    }

    @Test
    public void testInvalidPredicatesPattern() throws KuduException {
        final Map<String, String> rows = new HashMap<>();
        rows.put("key", "val1");
        rows.put("key1", "val1");
        
        String invalidPattern = "key1==val1";

        kuduScan.insertTestRecordsToKuduTable(DEFAULT_TABLE_NAME, rows);

        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
        runner.setProperty(ScanKudu.PREDICATES, invalidPattern);
        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "column1");
        runner.setValidateExpressionUsage(false);
        runner.enqueue("trigger flow file");
        runner.assertNotValid();

        runner.assertTransferCount(ScanKudu.REL_FAILURE, 0);
        runner.assertTransferCount(ScanKudu.REL_SUCCESS, 0);
        runner.assertTransferCount(ScanKudu.REL_ORIGINAL, 0);
    }
```